lte/gateway/c/core/oai/test/amf/test_amf_stateless.cpp

```diff
@@ -121,6 +125,210 @@ TEST(TestAMFStateConverter, TestStateToProto) {
+  ue_m5gmm_context1.mm_state                     = REGISTERED_IDLE;
+
+  ue_m5gmm_context1.sctp_assoc_id_key = 1;
+  ue_m5gmm_context1.gnb_ue_ngap_id    = 0x09;
+  ue_m5gmm_context1.gnb_ngap_id_key   = INVALID_GNB_UE_NGAP_ID_KEY;
+
+  ue_m5gmm_context1.amf_context.apn_config_profile.nb_apns = 1;
+  strncpy(
+      ue_m5gmm_context1.amf_context.apn_config_profile.apn_configuration[0]
+          .service_selection,
+      "internet", 8);
+  ue_m5gmm_context1.amf_context.apn_config_profile.apn_configuration[0]
+      .service_selection_length = 8;
+
+  ue_m5gmm_context1.amf_teid_n11 = 0;
+
+  ue_m5gmm_context1.amf_context.subscribed_ue_ambr.br_unit = KBPS;
+  ue_m5gmm_context1.amf_context.subscribed_ue_ambr.br_ul   = 1000;
+  ue_m5gmm_context1.amf_context.subscribed_ue_ambr.br_dl   = 10000;
+
+  ue_m5gmm_context1.paging_context.paging_retx_count = 0;
+
+  AmfNasStateConverter::ue_m5gmm_context_to_proto(
+      &ue_m5gmm_context1, &ue_context_proto);
+
+  AmfNasStateConverter::proto_to_ue_m5gmm_context(
+      ue_context_proto, &ue_m5gmm_context2);
+
+  EXPECT_EQ(
+      ue_m5gmm_context1.ue_context_rel_cause.present,
+      ue_m5gmm_context2.ue_context_rel_cause.present);
+  EXPECT_EQ(ue_m5gmm_context1.ecm_state, ue_m5gmm_context2.ecm_state);
+  EXPECT_EQ(ue_m5gmm_context1.mm_state, ue_m5gmm_context2.mm_state);
+
+  EXPECT_EQ(
+      ue_m5gmm_context1.sctp_assoc_id_key, ue_m5gmm_context2.sctp_assoc_id_key);
+  EXPECT_EQ(ue_m5gmm_context1.gnb_ue_ngap_id, ue_m5gmm_context2.gnb_ue_ngap_id);
+  EXPECT_EQ(
+      ue_m5gmm_context1.gnb_ngap_id_key, ue_m5gmm_context2.gnb_ngap_id_key);
+
+  EXPECT_EQ(
+      ue_m5gmm_context1.amf_context.apn_config_profile.nb_apns,
+      ue_m5gmm_context2.amf_context.apn_config_profile.nb_apns);
+
+  std::string str_in =
+      ue_m5gmm_context1.amf_context.apn_config_profile.apn_configuration[0]
+          .service_selection;
+  std::string str_out =
+      ue_m5gmm_context2.amf_context.apn_config_profile.apn_configuration[0]
+          .service_selection;
+  EXPECT_EQ(str_in, str_out);
+
+  EXPECT_EQ(ue_m5gmm_context1.amf_teid_n11, ue_m5gmm_context2.amf_teid_n11);
+
+  EXPECT_EQ(
+      ue_m5gmm_context1.amf_context.subscribed_ue_ambr.br_unit,
+      ue_m5gmm_context2.amf_context.subscribed_ue_ambr.br_unit);
+  EXPECT_EQ(
+      ue_m5gmm_context1.amf_context.subscribed_ue_ambr.br_ul,
+      ue_m5gmm_context2.amf_context.subscribed_ue_ambr.br_ul);
+  EXPECT_EQ(
+      ue_m5gmm_context1.amf_context.subscribed_ue_ambr.br_dl,
+      ue_m5gmm_context2.amf_context.subscribed_ue_ambr.br_dl);
+
+  EXPECT_EQ(
+      ue_m5gmm_context1.paging_context.paging_retx_count,
+      ue_m5gmm_context2.paging_context.paging_retx_count);
+}
+
+TEST(TestAMFStateConverter, TestAmfContextStateToProto) {
+#define AMF_CAUSE_SUCCESS 1
+  amf_context_t amf_ctx1 = {}, amf_ctx2 = {};
+  magma::lte::oai::EmmContext emm_context_proto = magma::lte::oai::EmmContext();
+
+  amf_ctx1.imsi64             = 222456000000101;
```

Comment:
Why is this value different from the digits of amf_ctxt1.imsi.u in lines 209-223 ?

Reply:
Comment Addressed